processor/deltatocumulativeprocessor/internal/data/add.go

```diff
@@ -23,7 +37,101 @@ func (dp Histogram) Add(in Histogram) Histogram {
 	panic("todo")
 }
 
-// nolint
 func (dp ExpHistogram) Add(in ExpHistogram) ExpHistogram {
-	panic("todo")
+	switch {
+	case dp.Timestamp() >= in.Timestamp():
+		fail("out of order")
+	case dp.Scale() != in.Scale():
+		fail("scale changed")
+	case dp.ZeroCount() != in.ZeroCount():
+		fail("zero count changed")
+	}
+
+	aggregate := func(dpBuckets, inBuckets pmetric.ExponentialHistogramDataPointBuckets) {
+		var (
+			dp   = Buckets{data: dpBuckets.BucketCounts(), offset: int(dpBuckets.Offset())}
+			in   = Buckets{data: inBuckets.BucketCounts(), offset: int(inBuckets.Offset())}
+			aggr = Buckets{data: pcommon.NewUInt64Slice()}
+		)
+		aggr.offset = int(min(dpBuckets.Offset(), inBuckets.Offset()))
+		if aggr.offset == dp.offset {
+			aggr.data = dp.data
+		}
+		aggr.EnsureLen(max(dp.Len(), in.Len()))
```

Comment:
I don't get this. Imagine `dp` has buckets `10 .. 20` and `in` has buckets `90 .. 100`. Then `aggr` would have to cover `10 .. 100`.